File: op/lab3/python/lab3.py
import math

lnPrevious = 0 
ln = 0
e = 0.000001
n=1
x=float(input("Enter value of the \'x\': "))
lnPrevious=(-1)**(n-1) * ((x**n)/n)
n+=1
ln = lnPrevious +  (-1)**(n-1) * (((x)**n)/n)
n+=1
while abs(ln-lnPrevious) > e :
    lnPrevious = ln
    ln += (-1)**(n-1) * (((x)**n)/n)
    n+=1
print(ln)

File: op/lab3/c++/lab3.cpp
#include<iostream>
#include<math.h>
using namespace std; 



int main(){
    float x, lnPrevious=0,ln=0, e = 0.000001;
    int n=1;
    cout << "Enter valou of the \'x\': ";
    cin >> x;
    lnPrevious = pow(-1,(n-1)) * float(pow(x,n))/n;
    n++;

    ln = lnPrevious +  pow(-1,(n-1)) * float(pow(x,n))/n;
    n++;
    while ( abs(ln - lnPrevious) > e) {
        lnPrevious = ln;
        ln +=  pow(-1,(n-1)) * float(pow(x,n))/n;
        cout << endl<< lnPrevious ;
        n++;
    }  
 
    return 0;}